main.ts

```diff
@@ -509,31 +547,37 @@ export default class VimrcPlugin extends Plugin {
 
 			let cmEditor = this.getCodeMirror(this.getActiveView());
 			// See https://codemirror.net/doc/manual.html#vimapi_events for events.
-			CodeMirror.on(cmEditor, "vim-keypress", async (vimKey: any) => {
-				if (vimKey != "<Esc>") { // TODO figure out what to actually look for to exit commands.
-					this.currentKeyChord.push(vimKey);
-					if (this.customVimKeybinds[this.currentKeyChord.join("")] != undefined) { // Custom key chord exists.
-						this.currentKeyChord = [];
-					}
-				} else {
-					this.currentKeyChord = [];
-				}
+			cmEditor.off('vim-keypress', this.onVimKeypress);
+			cmEditor.on('vim-keypress', this.onVimKeypress);
+			cmEditor.off('vim-command-done', this.onVimCommandDone);
+			cmEditor.on('vim-command-done', this.onVimCommandDone);
+		}
+	}
 
-				// Build keychord text
-				let tempS = "";
-				for (const s of this.currentKeyChord) {
-					tempS += " " + s;
-				}
-				if (tempS != "") {
-					tempS += "-";
-				}
-				this.vimChordStatusBar.setText(tempS);
-			});
-			CodeMirror.on(cmEditor, "vim-command-done", async (reason: any) => { // Reset display
-				this.vimChordStatusBar.setText("");
+	onVimKeypress = async (vimKey: any) => {
+		if (vimKey != "<Esc>") { // TODO figure out what to actually look for to exit commands.
+			this.currentKeyChord.push(vimKey);
+			if (this.customVimKeybinds[this.currentKeyChord.join("")] != undefined) { // Custom key chord exists.
 				this.currentKeyChord = [];
-			});
+			}
+		} else {
+			this.currentKeyChord = [];
+		}
+
+		// Build keychord text
+		let tempS = "";
+		for (const s of this.currentKeyChord) {
+			tempS += " " + s;
+		}
+		if (tempS != "") {
+			tempS += "-";
 		}
+		this.vimChordStatusBar.setText(tempS);
+	}
+
+	onVimCommandDone = async (reason: any) => {
+		this.vimChordStatusBar.setText("");

```

Comment:
This throws an exception if chord display is off

Reply:
fixed.